test/Plutip.purs

```diff
@@ -331,35 +362,27 @@ suite = do
           , BigInt.fromInt 100_000_000
           ]
       runPlutipContract config distribution \alice -> do
-        withKeyWallet alice do
-          pkh <- liftedM "Failed to get own PKH" ownPaymentPubKeyHash
-          let
-            constraints :: Constraints.TxConstraints Void Void
-            -- In real contracts, library users most likely want to use
-            -- `mustPayToPubKeyAddres` (we're not doing that because Plutip
-            -- does not provide stake keys).
-            constraints = Constraints.mustPayToPubKey pkh
-              $ Value.lovelaceValueOf
-              $ BigInt.fromInt 2_000_000
+        assertCorrectDistribution [ alice /\ distribution ]
+        assertNoUtxosAtBaseAddress alice
+        withKeyWallet alice signMultipleContract
 
-            lookups :: Lookups.ScriptLookups Void
-            lookups = mempty
-
-          ubTx1 <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
-          ubTx2 <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
-
-          withBalancedAndSignedTxs [ ubTx1, ubTx2 ] $ \txs -> do
-            locked <- getLockedInputs
-            logInfo' $ "Locked inputs inside bracket (should be nonempty): " <>
-              show
-                locked
-            traverse_ submitAndLog txs
+    -- TODO: not sure what's causing this to fail
```

Comment:
From slack

> (me) I added [this test](https://github.com/Plutonomicon/cardano-transaction-lib/blob/c3ad6a7c648c791c2adcefbdc0ea21731d9816b1/test/Plutip.purs#L358) for running the sign multiple contract with a stake key, but it ends up failing [here](https://github.com/Plutonomicon/cardano-transaction-lib/blob/c3ad6a7c648c791c2adcefbdc0ea21731d9816b1/test/Plutip.purs#L417) with "locked inputs map is not empty", while the test above it succeeds and the only difference is it doesn't use a stake key. I'm not sure why using a staking key is causing this, it seems unrelated to me, do you have any ideas?

> (@klntsky) So these locked inputs remain from the transactions that distribute balances, meaning our assertion is wrong.
I think we can be sure we are not going to try spending these already spent UTxOs, so feel free to clear locked inputs Ref manually  after sending the tx that distributes funds. See getLockedInputs - we want to set cache to mempty

Reply:
@klntsky If the assertion is wrong and the cache is as expected, could I just remove the assertion?

Reply:
@rynoV 

I think we should not remove the assertion and instead clear locked inputs. Because distribution of funds is transparent to the user, and since we await the tx, we can be sure the state of these utxos is settled. This locked inputs mechanism is designed to prevent double spending attempts in some scenarios. But there is nothing we should guard against in this case.